src/zarr/v3/group.py

```diff
@@ -271,8 +290,32 @@ def __repr__(self):
     async def nchildren(self) -> int:
         raise NotImplementedError
 
-    async def children(self) -> AsyncIterator[AsyncArray, AsyncGroup]:
-        raise NotImplementedError
+    async def children(self) -> AsyncGenerator[AsyncArray, AsyncGroup]:
+        """
+        Returns an AsyncGenerator over the arrays and groups contained in this group.
+        This method requires that `store_path.store` supports directory listing.
+        """
+        if not self.store_path.store.supports_listing:
+            msg = (
+                f"The store associated with this group ({type(self.store_path.store)}) "
+                "does not support listing, "
+                "specifically via the `list_dir` method. "
+                "This function requires a store that supports listing."
+            )
+
+            raise ValueError(msg)
+        subkeys = await self.store_path.store.list_dir(self.store_path.path)
+        # would be nice to make these special keys accessible programmatically,
+        # and scoped to specific zarr versions
+        subkeys_filtered = filter(lambda v: v not in ("zarr.json", ".zgroup", ".zattrs"), subkeys)
```

Comment:
```suggestion
        # and scoped to specific zarr versions
        subkeys_filtered = filter(lambda v: v not in ("zarr.json", ".zgroup", ".zarray"), subkeys)
```

Reply:
when will `.zarray` be an immediate subkey of a group? By contrast, we do expect to find `.zattrs` under the prefix of any v2 group with attributes.